core/src/main/java/org/apache/carbondata/core/scan/executor/impl/AbstractQueryExecutor.java

```diff
@@ -476,7 +476,11 @@ private BlockExecutionInfo getBlockExecutionInfoForBlock(QueryModel queryModel,
     blockExecutionInfo
         .setTotalNumberDimensionToRead(
             segmentProperties.getDimensionOrdinalToChunkMapping().size());
-    blockExecutionInfo.setPrefetchBlocklet(!queryModel.isReadPageByPage());
+    if (queryModel.isReadPageByPage()) {
```

Comment:
It seems this modification is not related with the description of this PR

Reply:
Updated description